### PR TITLE
Mihai/api docs refactor

### PR DESCRIPTION
**Pull request by iftodimihail**

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/112#" title="Author self-approved">iftodimihail</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: packages/api/src/templates/docs.js [hunk 1]
```diff
@@ -37,32 +37,44 @@ export default function MDXLayout({ data = {}, location }) {
...
         .reduce((propertiesAcc, property) => {
           const definitionProperty = defObject.properties[property];
-          const definitionPropertyRef = definitionProperty?.$ref || definitionProperty?.items?.$ref
+          const definitionPropertyRef = definitionProperty?.$ref || definitionProperty?.items?.$ref;
 
-          return definitionPropertyRef ?
-            // if the property contains a ref, call again extractDefinition
-            {
+          // if the property contains a ref, call again extractDefinition
+          if(definitionPropertyRef) {
```
Comment (codrin-iftimie) [inline packages/api/src/templates/docs.js:51]: We should add the eslint to the project ;)

File: packages/api/src/templates/docs.js [hunk 2]
```diff
@@ -37,32 +37,44 @@ export default function MDXLayout({ data = {}, location }) {
...
-                  // otherwise render the property type
-                  definitionProperty?.properties ?
-                    renderProperties(definitionProperty) :
-                    definitionProperty.type
+            });
+          } else {
+            // if property value is an array, render what type the elements are
+            if(definitionProperty.type === "array") {
+              return ({
+                ...propertiesAcc,
```
Comment (codrin-iftimie) [inline packages/api/src/templates/docs.js:62]: Let's avoid spreads if they are not needed. They can be quite slow.

File: packages/api/src/components/Swagger.js [hunk 3]
```diff
@@ -110,8 +111,88 @@ const normalizePath = path => {
...
+  return (
+    <Summary>
+      <Label>{label}:</Label> {value || `No ${label}`}
+    </Summary>
+  )
+}
+
+function Parameters({parameters, method, path}) {
+  return (
+    parameters?.length > 0 ? (
```
Comment (codrin-iftimie) [inline packages/api/src/components/Swagger.js:124]: maybe `if (!params?.length) { return <Summary> etc}` is more readable

File: packages/api/src/components/Swagger.js [hunk 4]
```diff
@@ -110,8 +111,88 @@ const normalizePath = path => {
...
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Type</th>
+              <th>Description</th>
+              <th>Required</th>
+            </tr>
+          </thead>
+          <tbody>
+            {parameters.map(parameter => (
```
Comment (codrin-iftimie) [inline packages/api/src/components/Swagger.js:137]: map functions should be declared to make them easier to read

File: packages/api/src/templates/docs.js (hunk 2 above)
Comment (iftodimihail) [reply packages/api/src/templates/docs.js:62]: well everywhere used in this function it is needed

**Pull request merged**